pkg/virt-controller/watch/migration.go

```diff
@@ -357,9 +361,13 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 		}
 	}
 
-	if !reflect.DeepEqual(migration.Status, migrationCopy.Status) ||
-		!reflect.DeepEqual(migration.Finalizers, migrationCopy.Finalizers) {
-		_, err := c.clientset.VirtualMachineInstanceMigration(migration.Namespace).Update(migrationCopy)
+	if !reflect.DeepEqual(migration.Status, migrationCopy.Status) {
```

Comment:
this is a change in behavior. 

shouldn't it now be

```
if status not equal {
    update status
}

if finalizers not equal {
    update finalizers
}
```

the way your logic is, we are only doing one or the other, not both.  was that intentional?

Reply:
I could change it. My idea was that in one reconcile loop it goes to the final state and on the next one it removes the finalizer.

Reply:
Made up my mind. I implemented the change in the status updater logic. It makes it even more complex if I pass back the new object too to directly use it for further actions. I would prefer to keep it this way. What do you think?

Reply:
wfm